astro/src/content/docs/get-started/start-here/step-5.mdx

```diff
@@ -0,0 +1,102 @@
+
+Our example app handles refresh tokens automatically. Let's look at how this is accomplished.
+
+Open the file `src/sdk.ts` and locate the function named `logInUser`. This function is where the access token, refresh token, and id token (part of the OpenID Connect specification) are written out to the browser as cookies. This function looks like this:
+
+```typescript
+logInUser(accessToken: AccessToken, res: Response) {
+  res.cookie(this.configuration.accessTokenCookieName, accessToken.access_token, { httpOnly: true });
+  res.cookie(this.configuration.idTokenCookieName, JSON.stringify(jose.decodeJwt(accessToken.id_token)), { httpOnly: false });
+
+  if (this.configuration.enableRefreshTokens && accessToken.refresh_token) {
+    res.cookie(this.configuration.refreshTokenCookieName, accessToken.refresh_token, { httpOnly: true });
+  }
+}
+```
+
+You can see that the access and refresh token are written out as `HttpOnly`. Our example application doesn't run over TLS, therefore we can't set the `Secure` flag. But for production applications, you should use TLS and mark cookies as `Secure`.
+
+Next, let's look at how the refresh token is used when the access token expires. Scroll to the `getUser` function in this file. This function loads the same cookie that was written out above. We are using the `jose` library, which throws an exception when the access token has expired. This exception is handled in the `catch` block like this:
+
+```typescript
+try {
+  payload = (await jose.jwtVerify(accessToken, this.JWKS, {
+    issuer: this.configuration.oauthIssuer,
+    audience: this.configuration.applicationId,
+  })).payload;
+} catch (e) {
+  // NOTE: Check if the JWT is expired
+  if (e instanceof jose.errors.JWTExpired) {
+    // ...
+  }
+}
+```
+
+Inside this `catch` block, we check if the exception is due to an expired JWT. If the JWT is expired, we refresh it using this code:
+
+```typescript
+// Refreshing is disabled, so the user is logged out
+if (!this.configuration.enableRefreshTokens) {
+  return null;
+}
+
+// Load the refresh token from the cookie
+let refreshToken = req.cookies[this.configuration.refreshTokenCookieName];
+if (!refreshToken) {
+  return null;
+}
+
+// Try refreshing the token
+let response = await this.refreshToken(refreshToken);
+if (!response) {
+  return null;
+}
+
+// Update the cookies making the assumption that they are both valid since we just got them from FusionAuth
+accessToken = response.access_token;
+res.cookie(this.configuration.accessTokenCookieName, accessToken, { httpOnly: true });
+
+if (this.configuration.enableRefreshTokens && response.refresh_token) {
```

Comment:
Why do we need `this.configuration.enableRefreshTokens` here? We just checked that on line 61.

Reply:
True. This is legacy and I just copy and pasted things around. I'll fix that on the example app and here.